client/state/sites/reducer.js

```diff
@@ -204,6 +205,17 @@ export function items( state = {}, action ) {
 			}, state );
 		}
 
+		case SITE_DOMAINS_PRIMARY_SET_COMPLETED: {
+			const { siteId, primaryDomainName } = action;
+			return {
+				...state,
+				[ siteId ]: {
+					...state[ siteId ],
+					URL: `http://${ primaryDomainName }`
```

Comment:
It would require more operations on site object to make it work.

I'll explore server side solution for this one. I'm thinking about sending updated site object from sever upon primary domain change.